### PR TITLE
Error handler

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Right now there is no way to handle exceptions while a Route is being triggered.
We want to map a callable to be resolved when a particular exception occurs. 

## 🔖 Changes

- Rename `MappingInterfaces` to `Bindings`
- Add `Handlers` which maps potential exceptions while running a Route which ends in an Exception
- When ussing `Router::configure(...)` the order or the parameters from that callable doesn't matter

### 🧪 Example
```php
# A
static function (Routes $routes, Bindings $bindings, Handlers $handlers) void { ...
# B
static function (Routes $routes, Handlers $handlers, Bindings $bindings) void { ...
# C
static function (Routes $routes, Handlers $handlers) void { ...
# D
static function (Routes $routes, Bindings $bindings) void { ...
```
> Any order, any arguments. This works by using reflextion at the callable on run time and figuring out which types are you using on different position, so it will use them properly. Kudos to @Tito-Kati 🧠

### CODE REVIEW

File: src/Router/Exceptions/UnhandledException.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gacela\Router\Exceptions;
+
+use RuntimeException;
+
+final class UnhandledException extends RuntimeException
```
Comment (JesusValeraDev) [inline src/Router/Exceptions/UnhandledException.php]: `Runtime` or `Exception` in this particular case?

File: src/Router/Router.php [hunk 2]
```diff
@@ -49,4 +52,18 @@ private static function findRoute(Routes $routes): Route
...
+    {
+        $handler = $handlers->getByException($exception);
+        if ($handler === null) {
+            header('HTTP/1.1 500 Internal Server Error');
+        } else {
+            /** @var mixed $result */
+            $result = $handler($exception);
+            if (is_string($result)) {
+                echo $result;
+            }
```
Comment (JesusValeraDev) [inline src/Router/Router.php]: Can the `$result` not be a string?

File: src/Router/Exceptions/UnhandledException.php (hunk 1 above)
Comment (antonio-gg-dev) [reply src/Router/Exceptions/UnhandledException.php]: This exception is only being used for testing, should it be part of the production code?

File: src/Router/Router.php (hunk 2 above)
Comment (antonio-gg-dev) [reply src/Router/Router.php]: Should not 🤔

**Review by antonio-gg-dev — APPROVED** [review]

**Pull request merged**